Add a parameter register for IS-04 capabilities (#21)
AMWA-TV/nmos-parameter-registers: capabilities/README.md

# NMOS Capabilities

> This Capabilities parameter register contains values that may be used to identify a capability, used in the 'caps' property of the resources defined in the [AMWA IS-04 NMOS Discovery and Registration Specification](https://github.com/AMWA-TV/nmos-discovery-registration).

Reviewer: ```suggestion
This Capabilities parameter register contains values that may be used to identify a capability, used in the 'caps' property of the resources defined in the [AMWA IS-04 NMOS Discovery and Registration Specification](https://specs.amwa.tv/is-04/).
```

- Each entry MUST have a short description.
- Each entry MUST identify any AMWA Specifications and versions from which it is valid.
- Additions and updates to this parameter register are to be submitted via a Pull Request (PR) according to the [General Procedures and Criteria](../README.md#general-procedures-and-criteria).

Query API clients MUST be tolerant to the presence of capabilities not yet defined here which may be added in later API versions.

Manufacturers MAY use their own namespaces to indicate capabilities which are not currently defined within the NMOS namespace (`urn:x-nmos:cap:`). In order to avoid collisions with simple names allocated by AMWA specifications, they MUST NOT use capability names that do not start with `urn:`.

Capabilities are most used by IS-04 Receivers to indicate what they may consume, but may be used by other 'caps' objects in the future, potentially to indicate what they may be re-configured to generate.

## Values

- **Name:** media_types
  - **Description:** Identifies Flows which may be created or consumed based upon their 'media_type' attribute.

>   - **Specification:** [AMWA IS-04 v1.1](https://github.com/AMWA-TV/nmos-discovery-registration/tree/v1.1.x)

Reviewer: ```suggestion
  - **Specification:** [AMWA IS-04 v1.1](https://specs.amwa.tv/is-04/v1.1.x/)
```

This should change to .../v1.1/ when that means the latest published patch release on the v1.1 branch, after v1.1 release becomes v1.1.0.

  - **Applicability:** AMWA IS-04 v1.1+
- **Name:** event_types
  - **Description:** Identifies Sources or Flows which may be created or consumed based upon their 'event_type' attribute.

>   - **Specification:** [AMWA IS-04 v1.3](https://github.com/AMWA-TV/nmos-discovery-registration/tree/v1.3-dev)

Reviewer: ```suggestion
  - **Specification:** [AMWA IS-04 v1.3](hhttps://specs.amwa.tv/is-04/v1.3.x/)
```

  - **Applicability:** AMWA IS-04 v1.3+
- **Name:** constraint_sets
  - **Description:** Identifies streams from Senders which may be created or consumed based upon the attributes of the associated Source or Flow or contents of the associated transport file.

>   - **Specification:** [AMWA BCP-004-01 (Work In Progress)](https://github.com/AMWA-TV/nmos-receiver-capabilities/tree/v1.0-dev)

Reviewer: ```suggestion
  - **Specification:** [AMWA BCP-004-01](https://specs.amwa.tv/bcp-004-01/v1.0-dev/)
```

  - **Applicability:** AMWA IS-04
  - **Note:** A JSON schema for Constraint Sets, supporting validation of all the Constraint Set Metadata and Parameter Constraints defined in this register, is available as **[constraint_set.json](constraint_set.json)**.
    It MAY be used in place of the file with the same name in the AMWA BCP-004-01 specification.
- **Name:** version
  - **Description:** String formatted TAI timestamp (<seconds>:<nanoseconds>) indicating when an attribute of the 'caps' object last changed.

>   - **Specification:** [AMWA BCP-004-01 (Work In Progress)](https://github.com/AMWA-TV/nmos-receiver-capabilities/tree/v1.0-dev)

Reviewer: ```suggestion
  - **Specification:** [AMWA BCP-004-01](https://specs.amwa.tv/bcp-004-01/v1.0-dev/)
```

  - **Applicability:** AMWA IS-04

### Constraint Set Metadata

- **Name:** urn:x-nmos:cap:meta:label
  - **Description:** Freeform string label to provide a human-readable name for a Constraint Set.

>   - **Specification:** [AMWA BCP-004-01 (Work In Progress)](https://github.com/AMWA-TV/nmos-receiver-capabilities/tree/v1.0-dev)

Reviewer: ```suggestion
  - **Specification:** [AMWA BCP-004-01](https://specs.amwa.tv/bcp-004-01/v1.0-dev/)
```

  - **Applicability:** AMWA IS-04
- **Name:** urn:x-nmos:cap:meta:preference
  - **Description:** Expresses the relative 'weight' that the Receiver assigns to its preference for the streams satisfied by the associated Constraint Set.

>   - **Specification:** [AMWA BCP-004-01 (Work In Progress)](https://github.com/AMWA-TV/nmos-receiver-capabilities/tree/v1.0-dev)

Reviewer: ```suggestion
  - **Specification:** [AMWA BCP-004-01](https://specs.amwa.tv/bcp-004-01/v1.0-dev/)
```

  - **Applicability:** AMWA IS-04
- **Name:** urn:x-nmos:cap:meta:enabled
  - **Description:** Indicates whether a Constraint Set is available to use immediately (true) or whether this is an offline capability which can be activated via some unspecified configuration mechanism (false).

>   - **Specification:** [AMWA BCP-004-01 (Work In Progress)](https://github.com/AMWA-TV/nmos-receiver-capabilities/tree/v1.0-dev)

Reviewer: ```suggestion
  - **Specification:** [AMWA BCP-004-01](https://specs.amwa.tv/bcp-004-01/v1.0-dev/)
```
